framework/src/play/src/main/scala/play/api/Configuration.scala

```diff
@@ -37,8 +41,9 @@ object Configuration {
         devSettings.get("config.file").orElse(Option(System.getProperty("config.file")))
           .map(f => new File(f)).getOrElse(new File(appPath, "conf/application.conf"))
       }
+      val playOverridesConfig: Config = getPlayOverridesConfig(Thread.currentThread().getContextClassLoader())
```

Comment:
@patriknw You are absolutely right! We should silence the config checker only for RP releases that include Play 2.4+. Play 2.3 and earlier can't be affected by the problem.